src/main/java/com/google/api/codegen/go/GoGapicContext.java

```diff
@@ -101,6 +101,34 @@ public GoGapicContext(Model model, ApiConfig apiConfig) {
     this.goCommon = new GoContextCommon();
   }
 
+  /**
+   * Returns the Go package name.
+   *
+   * Retrieved by splitting the package string on `/` and returning the second to last string if
+   * possible, and otherwise the last string in the resulting array.
+   */
+  public String getPackageName() {
```

Comment:
This is slightly ugly... the trick is getting the package name for the service to be `logging` instead of `apiv2`, and the only way I could think of is to pull the second to last string of the import path.